docs/example-confluence/src/confluenceParser.ts

```diff
@@ -0,0 +1,176 @@
+import * as TurndownService from 'turndown';
+
+interface HeadingData {
+    title: string;
+    hierarchy: string;
+    result: string;
+}
+
+interface HeadingStackItem {
+    level: number;
+    text: string;
+}
+
+export class ConfluenceDataParser {
+    private targetHeadingsLevel: number;
+    private pageName: string = '';
+    private turndownService: TurndownService;
+    private html: string;
+
+    /**
+     * Constructs a new ConfluenceDataParser instance.
+     * @param html The HTML content to parse
+     * @param pageName The name of the page
+     * @param targetHeadingsLevel The target heading level to extract
+     */
+    constructor(html: string, pageName: string, targetHeadingsLevel: number) {
```

Comment:
Modern JS/TS support instance variable declaration directly in the constructor. Let's use this one if there are no build problems

```suggestion
    private turndownService: TurndownService;

    /**
     * Constructs a new ConfluenceDataParser instance.
     * @param html The HTML content to parse
     * @param pageName The name of the page
     * @param targetHeadingsLevel The target heading level to extract
     */
    constructor(private readonly html: string, private readonly pageName: string, private readonly targetHeadingsLevel: number) {
```